observationtree/src/main/scala/explore/observationtree/ConstraintSetObsList.scala

```diff
@@ -304,39 +304,39 @@ object ConstraintSetObsList {
-                        val nextToSelect  = constraintSetsWithIdx.find(_._2 === csIdx + 1).map(_._1)
-                        val prevToSelect  = constraintSetsWithIdx.find(_._2 === csIdx - 1).map(_._1)
-                        val focusOnDelete = nextToSelect.orElse(prevToSelect)
-
-                        val csObs = obsByConstraintSet.get(csId.some).orEmpty
-
-                        val opIcon = csObs.nonEmpty.fold(
-                          Icon(
-                            "chevron " + props.expandedIds.get
-                              .exists(_ === csId)
-                              .fold("down", "right")
-                          )(^.cursor.pointer,
-                            ^.onClick ==> { e: ReactEvent =>
-                              e.stopPropagationCB >>
-                                toggleExpanded(csId, props.expandedIds).runAsyncCB
-                                  .asEventDefault(e)
-                                  .void
-                            }
-                          ),
-                          Icons.ChevronRight
-                        )
+              // Start constraint sets tree
+              ReflexElement(minSize = 36, clazz = ExploreStyles.ObsTreeSection)(
+                Header(block = true, clazz = ExploreStyles.ObsTreeHeader)("CONSTRAINTS"),
+                <.div(ExploreStyles.ObsTree)(
+                  <.div(ExploreStyles.ObsScrollTree)(
+                    constraintSetsWithIdx.toTagMod { case (constraintSet, csIdx) =>
+                      val csId          = constraintSet.id
+                      val nextToSelect  = constraintSetsWithIdx.find(_._2 === csIdx + 1).map(_._1)
+                      val prevToSelect  = constraintSetsWithIdx.find(_._2 === csIdx - 1).map(_._1)
+                      val focusOnDelete = nextToSelect.orElse(prevToSelect)
+
+                      val csObs = obsByConstraintSet.get(csId.some).orEmpty
+
+                      val opIcon = csObs.nonEmpty.fold(
+                        Icon(
+                          "chevron " + props.expandedIds.get
+                            .exists(_ === csId)
+                            .fold("down", "right")
+                        )(^.cursor.pointer,
+                          ^.onClick ==> { e: ReactEvent =>
+                            e.stopPropagationCB >>
+                              toggleExpanded(csId, props.expandedIds).runAsyncCB
+                                .asEventDefault(e)
+                                .void
+                          }
+                        ),
+                        Icons.ChevronRight
+                      )
 
-                        val obsSelected = props.focused.get
-                          .exists(f => csObs.map(obs => FocusedObs(obs.id)).exists(f === _))
+                      val obsSelected = props.focused.get
+                        .exists(f => csObs.map(obs => FocusedObs(obs.id)).exists(f === _))
 
+                      <.span(^.key := csId.toString)(
```

Comment:
Do we need this `span` along with the other changes you made?

Reply:
Good catch. I was just trying things out here. I'll remove it, it's not necessary.